app/components/nav/user-nav.tsx

```diff
@@ -0,0 +1,82 @@
+import {
+  Avatar,
+  AvatarFallback,
+  AvatarImage,
+} from "@/app/components/ui/avatar";
+import { Button } from "@/app/components/ui/button";
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuGroup,
+  DropdownMenuItem,
+  DropdownMenuLabel,
+  DropdownMenuSeparator,
+  DropdownMenuShortcut,
+  DropdownMenuTrigger,
+} from "@/app/components/ui/dropdown-menu";
+import { UserIcon } from "lucide-react";
+import { Session } from "next-auth";
+import { signOut } from "next-auth/react";
+import Link from "next/link";
+
+export function UserNav({ session }: { session: Session }) {
+  const {
+    user: { id, email, role, staffProfile, facilityProfile },
+  } = session;
+  return (
+    <DropdownMenu>
+      <DropdownMenuTrigger asChild>
+        <Button variant="ghost" className="relative h-8 w-8 rounded-full">
+          <Avatar className="h-9 w-9">
+            <AvatarImage
+              src={
+                role === "FACILITY"
+                  ? facilityProfile?.profileImage
+                  : role === "STAFF"
+                  ? staffProfile?.profileImage
+                  : "/avatar"
+              }
```

Comment:
This part seems a bit confusing, lets move that logic above and simplify the code here

```suggestion
              src={getSrc(role)}
```